server/graphql/v2/mutation/ApplicationMutations.js

```diff
@@ -21,21 +22,24 @@ const createApplication = {
       throw new Unauthorized('You need to be authenticated to create an application.');
     }
 
-    const numberOfAppsForThisUser = await models.Application.count({
-      where: {
-        CollectiveId: req.remoteUser.CollectiveId,
-      },
-    });
+    const collective = args.application.account
+      ? await fetchAccountWithReference(args.application.account, { throwIfMissing: true })
+      : req.remoteUser.collective;
+
+    if (!req.remoteUser.isAdminOfCollective(collective)) {
+      throw new Forbidden();
+    }
 
-    if (numberOfAppsForThisUser >= config.limits.maxNumberOfAppsPerUser) {
+    const numberOfAppsForThisAccount = await models.Application.count({ where: { CollectiveId: collective.id } });
+    if (numberOfAppsForThisAccount >= config.limits.maxNumberOfAppsPerUser) {
```

Comment:
We should probably rename `maxNumberOfAppsPerUser` to `maxNumberOfAppsPerAccount`